portal/src/graphql/adminapi/AddUserScreen.tsx

```diff
@@ -77,11 +77,30 @@ function isPasswordNeeded(
     return false;
   }
 
+  const enablePasswordField = (authenticator: PrimaryAuthenticatorType) => {
```

Comment:
Actually, should we always enable password field is password authenticator is enabled?
@fungc-io

Reply:
I don't think the password is needed if it's not compulsory. But why was @Ngkaokis experiencing the error if pw is not needed?

Is this solved by changing the design to

> Email
> [Email input]
>
> Set password for this user?
> [Toggle: Yes/No - default "No"]
>
> <-- if set pw is yes -->
> Password
> [ Password input ]
> </ --end if -- >

Reply:
It occurs when we enable both passwordless and password authenticator, and prioritize password authenticator first.

Front end side disable the password field as long as we enable passwordless authenticator.

But backend side raise error when password authenticator is the first one and no password is given.

I fixed it but syncing the password field disable logic as the backend one.

Reply:
So the criteria of enabled password field should be the first primary authenticator is "password", and not related to other authenticators?

Reply:
There maybe a situation that

1. Passwordless by email
2. Password
3. Passwordless by phone

In such situation, create user with email doesn't need password but create with phone does need. 

So need to check the order as well

Reply:
The problem here is actually:
1. When create user by email, the possibly created primary authenticators are `["oob_otp_email", "password"]`. And in this case "password" is not the first primary authenticator.
2. When create user by phone, the possibly created primary authenticators are `["password", "oob_otp_sms"]`, so "password" becomes the first primary authenticator.

So I suggest to modify this part so that it is easier to understand the logic behind:

```typescript
fuction filterAuthenticators(allowedTypes: PrimaryAuthenticatorType[]):  PrimaryAuthenticatorType[] {
  // TODO: Filter primary authenticators by the provided `allowedTypes`
}

// ... In isPasswordNeeded
const relatedAuthenticators: PrimaryAuthenticatorType[]
switch (loginIdKeySelected) {
  case "email":
    relatedAuthenticators = filterAuthenticators(["oob_otp_email", "password", "passkey"])
  case "phone":
    relatedAuthenticators = filterAuthenticators(["oob_otp_sms", "password", "passkey"])
  case "username":
    relatedAuthenticators = filterAuthenticators(["password", "passkey"])
  default:
    relatedAuthenticators = filterAuthenticators([])
}

return relatedAuthenticators.length > 0 && relatedAuthenticators[0] === "password"
```